packages/shared/scope-hoisting/src/concat.js

```diff
@@ -195,87 +192,10 @@ function parse(code, sourceFilename) {
   return ast.program.body;
 }
 
-function getUsedExports(
-  bundle: NamedBundle,
-  bundleGraph: BundleGraph<NamedBundle>,
-): Map<string, Set<Symbol>> {
-  let usedExports: Map<string, Set<Symbol>> = new Map();
-
-  let entry = bundle.getMainEntry();
-  if (entry) {
-    for (let {asset, symbol} of bundleGraph.getExportedSymbols(entry)) {
-      if (symbol) {
-        markUsed(asset, symbol);
-      }
-    }
-  }
-
-  bundle.traverseAssets(asset => {
-    for (let dep of bundleGraph.getDependencies(asset)) {
-      let resolvedAsset = bundleGraph.getDependencyResolution(dep, bundle);
-      if (!resolvedAsset) {
-        continue;
-      }
-
-      for (let [symbol, {local}] of dep.symbols) {
-        if (local === '*') {
-          continue;
-        }
-
-        if (symbol === '*') {
-          for (let {asset, symbol} of bundleGraph.getExportedSymbols(
-            resolvedAsset,
-          )) {
-            if (symbol) {
-              markUsed(asset, symbol);
-            }
-          }
-        }
-
-        markUsed(resolvedAsset, symbol);
-      }
-    }
-
-    // If the asset is referenced by another bundle, include all exports.
-    if (bundleGraph.isAssetReferencedByDependant(bundle, asset)) {
-      markUsed(asset, '*');
-      for (let {asset: a, symbol} of bundleGraph.getExportedSymbols(asset)) {
-        if (symbol) {
-          markUsed(a, symbol);
-        }
-      }
-    }
-  });
-
-  function markUsed(asset, symbol) {
-    let resolved = bundleGraph.resolveSymbol(asset, symbol);
-
-    let used = usedExports.get(resolved.asset.id);
-    if (!used) {
-      used = new Set();
-      usedExports.set(resolved.asset.id, used);
-    }
-
-    used.add(resolved.exportSymbol);
-  }
-
-  return usedExports;
-}
-
-function shouldSkipAsset(
-  bundleGraph: BundleGraph<NamedBundle>,
-  asset: Asset,
-  usedExports: Map<string, Set<Symbol>>,
-) {
+function shouldSkipAsset(bundleGraph: BundleGraph<NamedBundle>, asset: Asset) {
   return (
     asset.sideEffects === false &&
-    !asset.meta.isCommonJS &&
-    (!usedExports.has(asset.id) ||
-      nullthrows(usedExports.get(asset.id)).size === 0) &&
-    !bundleGraph.getIncomingDependencies(asset).find(d =>
-      // Don't exclude assets that was imported as a wildcard
-      d.symbols.hasExportSymbol('*'),
-    )
+    bundleGraph.getUsedSymbolsAsset(asset).size == 0
```

Comment:
There are two ways an asset(subgraph) can be excluded:
- the whole subgraph can be skipped if all dependencies to it are unused (= no used symbols on `dependency.usedSymbols`), so this would be some unused import/reexport
- a single asset can be skipped if none of its symbols are used (`asset.usedSymbols`), so this would be a file which has only reexports and can be excluded while its children are still needed